src/uwtools/config/core.py

```diff
@@ -755,19 +682,153 @@ def print_config_section(config: dict, key_path: List[str]) -> None:
     key=value pairs, one per line.
     """
     keys = []
+    current_path = "<unknown>"
     for section in key_path:
         keys.append(section)
         current_path = " -> ".join(keys)
         try:
             subconfig = config[section]
-        except KeyError:
-            _log_and_error(f"Bad config path: {current_path}")
+        except KeyError as e:
+            raise _log_and_error(f"Bad config path: {current_path}") from e
         if not isinstance(subconfig, dict):
-            _log_and_error(f"Value at {current_path} must be a dictionary")
+            raise _log_and_error(f"Value at {current_path} must be a dictionary")
         config = subconfig
     output_lines = []
     for key, value in config.items():
         if type(value) not in (bool, float, int, str):
-            _log_and_error(f"Non-scalar value {value} found at {current_path}")
+            raise _log_and_error(f"Non-scalar value {value} found at {current_path}")
         output_lines.append(f"{key}={value}")
     print("\n".join(sorted(output_lines)))
+
+
+def realize_config(
+    input_file: OptionalPath,
+    input_format: str,
+    output_file: OptionalPath,
+    output_format: str,
+    values_file: OptionalPath,
+    values_format: Optional[str],
+    values_needed: bool = False,
+    dry_run: bool = False,
```

Comment:
This block of arguments is repeated in several places and might be a good use case for a data class as envisioned by [UW-342](https://jira.epic.oarcloud.noaa.gov/browse/UW-342).